src/tainting/Taint.ml

```diff
@@ -34,12 +34,21 @@ type tainted_tokens = G.tok list [@@deriving show]
 type 'a call_trace =
   | PM of PM.t * 'a
   | Call of G.expr * tainted_tokens * 'a call_trace
+  (* We keep this as part of a "call trace" in order to denote when
+     an incoming source has been propagated to have a different taint label.
+     We need this so that we can compute the new label of a propagated source,
+     but so we can remember what the original source label was as well.
+     For all intents and purposes, `Propagate` should only be visible in the
+     case where we are calling [label_of_source_trace].
+  *)
+  | Propagate of 'a call_trace * string (* label *)
```

Comment:
I think it should be `orig` or `taint` that should be extended rather than `call_trace`.

Reply:
i think a `label: string` field for `taint` probably shouldn't work, because `Arg` as a source is "polymorphic" in its label, is that correct? i can just fall back to changing the `Src source` label

Reply:
hm. actually.

Reply:
i think there's a general issue with taint summaries as they relate to taint labels, that we should address first prior to talking about propagators

Reply:
i know you already identified there were some regressions with pro w.r.t. taint labels but i only just started thinking about it

Reply:
for context im thinking of an example that looks like this:
```
public class Foo {
  public void foo(int a) {
    b = convert(a); 

    sink(b);
  }

  public void main() {
    foo("a");
  }
}
```

```
rules:
- id: taint-labels-interproc
  message: |
    we should be able to do interproc taint findings involving labels!
  severity: ERROR
  languages: [java]
  mode: taint
  pattern-sources:
    - pattern: |
        "a"
      label: A
    - pattern: |
        convert(...)
      label: B
      requires: A
  pattern-sinks:
    - pattern: |
        sink(...)
      requires: B
 ```
 
 (you may be aware of this already)

Reply:
Yep, I'm aware that taint labels don't work inter-procedurally. I'm fine merging this PR if we replace the `Propagate` constructor with a `label: string` field in `taint` for example, I don't think it will break anything that currently works. There is work to do to update taint signatures and taint traces to work with labels.